shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -484,25 +533,33 @@ private void copyFromLocalFile(AlluxioURI srcPath, AlluxioURI dstPath)
    *
    * @param srcPath the {@link AlluxioURI} of the source file in the local filesystem
    * @param dstPath the {@link AlluxioURI} of the destination
+   * @throws InterruptedException when failed to send messages to the pool
    */
   private void asyncCopyLocalPath(CopyThreadPoolExecutor pool, AlluxioURI srcPath,
-      AlluxioURI dstPath) throws AlluxioException, IOException {
+      AlluxioURI dstPath) throws InterruptedException {
     File src = new File(srcPath.getPath());
     if (!src.isDirectory()) {
       pool.submit(() -> {
         try {
           copyFromLocalFile(srcPath, dstPath);
-          pool.println(String.format("Copied %s to %s.", srcPath, dstPath));
-        } catch (Exception e) {
-          pool.println(e.getMessage());
+          pool.sendMessage(String.format("Copied %s to %s.", srcPath, dstPath));
+        } catch (AlluxioException | IOException e) {
+          pool.sendMessage(new IOException(String.format("Failed to copy %s to %s.", srcPath,
```

Comment:
the useful information will be in `e`, so could we include `e.toString()` in the message? Then the exception is still useful even without the full stack trace

Reply:
I refactored sendMessage API.